### PR TITLE
Route table syntax

### PR DESCRIPTION
**Pull request by mtnygard**
Table routing should be documented.

* Moved old routing guide to reflect that it is specific to the terse syntax.
* Wrote a new routing guide for the table based syntax. The style is very different. The new guide starts with the simplest possible thing and builds up from there.
* Added a section on debugging routes, including reference to a new function `io.pedestal.http.route/try-routing-for` that lets the user try routes out from REPL without involving all the HTTP plumbing.

### CODE REVIEW

**Comment by mtnygard** [comment]
This builds on PR #394.

File: guides/documentation/service-routing.md [hunk 1]
```diff
@@ -12,601 +12,360 @@
...
 
 ```clojure
-["/user" {:get list-users
-          :post add-user}
- ["/:user-id"
- ^:constraints {:user-id #"[0-9]+"}
-  {:put update-user}
-  [^:constraints {:view #"long|short"}
-   {:get view-user}]]]
+["/user/:user-id" :get view-user :constraints {:user-id #"[0-9]+"}
```
Comment (kirillsalykin) [inline guides/documentation/service-routing.md]: Probably misses `]`.

Comment (mtnygard) [inline guides/documentation/service-routing.md]: Thank you, good catch!

**Comment by ohpauleez** [comment]
Thanks so much!

**Pull request merged**